### PR TITLE
Update metatype descriptions for OpenAPI Endpoints

### LINKED ISSUE #25559: Review configuration descriptions
Review the metatype descriptions for each of the configuration properties.

- Currently the descriptions mostly restate the names and don't give the user much advice about how to decide what they put in there.
- It says "OpenAPI 3.0/Swagger documents" and I don't think that's correct.

https://github.com/OpenLiberty/open-liberty/blob/de1f27bc421061d63aab7dba0cd9c7eff1b483de/dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties#L23

### PR DESCRIPTION
**Pull request by abutch3r**
Update descriptions used for the new  OpenAPI Endpoints customization properties to be more informative about the properties and behavior if not set

Fixes #25559

### CODE REVIEW

**Comment by abutch3r** [comment]
#build

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -20,7 +20,7 @@ mpOpenAPI=MicroProfile OpenAPI
 mpOpenAPI.desc=Configuration for MicroProfile OpenAPI.
 
 docPath=MicroProfile OpenAPI document endpoint path
-docPath.desc=Path to obtain OpenAPI 3.0/Swagger documents.
+docPath.desc=Specifies the path for obtaining OpenAPI documents. If no value is specified, the path is set to /openapi
```
Comment (is273) [inline dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties]: Should have a period at the end of the sentence: If no value is specified, the path is set to /openapi.

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties [hunk 2]
```diff
@@ -20,7 +20,7 @@ mpOpenAPI=MicroProfile OpenAPI
...
 
 uiPath=MicroProfile OpenAPI UI endpoint path
-uiPath.desc=Path for the OpenAPI UI.
\ No newline at end of file
+uiPath.desc=Specifies the path for accessing the MicroProfile OpenAPI UI. If no value is specified, the path is set to $docPath/ui
```
Comment (is273) [inline dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties]: Should have a period at the end of the sentence: If no value is specified, the path is set to $docPath/ui.

**Comment by abutch3r** [comment]
#libby #build

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties (hunk 2 above)
Comment (abutch3r) [reply dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties]: Periods have now been added

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (abutch3r) [reply dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties]: Periods have now been added

**Comment by is273** [comment]
Review is done. ID reviewed

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties [hunk 3]
```diff
@@ -20,7 +20,7 @@ mpOpenAPI=MicroProfile OpenAPI
 mpOpenAPI.desc=Configuration for MicroProfile OpenAPI.
 
 docPath=MicroProfile OpenAPI document endpoint path
-docPath.desc=Path to obtain OpenAPI 3.0/Swagger documents.
+docPath.desc=Specifies the path for obtaining OpenAPI documents. If no value is specified, the path is set to /openapi.
 
 uiPath=MicroProfile OpenAPI UI endpoint path
-uiPath.desc=Path for the OpenAPI UI.
\ No newline at end of file
+uiPath.desc=Specifies the path for accessing the MicroProfile OpenAPI UI. If no value is specified, the path is set to $docPath/ui.
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties]: An example may be needed. I don't think it's clear that you're setting the path for the endpoint at which the document and UI are exposed. I'm not sure what the correct terminology to use is here.

**Comment by posmith** [comment]
L2 message review not required.

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties [hunk 4]
```diff
@@ -20,7 +20,7 @@ mpOpenAPI=MicroProfile OpenAPI
 mpOpenAPI.desc=Configuration for MicroProfile OpenAPI.
 
 docPath=MicroProfile OpenAPI document endpoint path
-docPath.desc=Path to obtain OpenAPI 3.0/Swagger documents.
+docPath.desc=Specifies the path for obtaining OpenAPI documents. If no value is specified, the path is set to /openapi.
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties]: If the default value is set in the metatype then we don't need to repeat it here in the description.

Comment (abutch3r) [reply dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties]: The metatype defines a default for the docPath only. the UI Path does not given its default is relative to the docPath. so should remove the second part from docPath, but should be kept by uiPath

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties (hunk 3 above)
Comment (abutch3r) [reply dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties]: here are some relatable examples that 
https://github.com/OpenLiberty/open-liberty/blob/324a3987f79ae4055d672ca9326776030dffe4c8/dev/com.ibm.ws.security.openidconnect.client/resources/OSGI-INF/l10n/metatype.properties#L32
This does include a bit more detail about how what is set is related to the whole server address

https://github.com/OpenLiberty/open-liberty/blob/324a3987f79ae4055d672ca9326776030dffe4c8/dev/com.ibm.ws.security.social/resources/OSGI-INF/l10n/metatype.properties#L192

Most other `Path` examples are for file paths or JDBC which are unrelated to URLs. replacing `path` within the description with `context path` might clarify what the path is for

**Comment by abutch3r** [comment]
#libby

**Comment by abutch3r** [comment]
#libby

**Review by Azquelt — APPROVED** [review]

**Comment by abutch3r** [comment]
#build

**Comment by abutch3r** [comment]
Build failure is for testHADBRecoverableStartupFailover which is in no way related to OpenAPI or the changes to the descirptors which would fail more tests if it did truly break something

**Pull request merged**